pkg/agent/openflow/pipeline.go

```diff
@@ -573,6 +576,30 @@ func (c *client) conntrackBypassRejectFlow(proto binding.Protocol) binding.Flow
 		Done()
 }
 
+// dnsResponseBypassConntrackFlow generates a flow which is used to bypass the
+// dns response packetout from conntrack, to avoid unexpected packet drop.
+func (c *client) dnsResponseBypassConntrackFlow() binding.Flow {
+	conntrackTable := c.pipeline[conntrackTable]
+	return conntrackTable.BuildFlow(priorityHigh).
+		MatchRegFieldWithValue(CustomReasonField, CustomReasonDNS).
+		Cookie(c.cookieAllocator.Request(cookie.Default).Raw()).
+		Action().ResubmitToTable(l2ForwardingCalcTable).
+		Done()
+}
+
+// dnsResponseBypassPacketInFlow generates a flow which is used to bypass the
+// dns packetIn conjunction flow for dns response packetOut. This packetOut
+// should be sent directly to the requesting client without being intercepted
+// again.
+func (c *client) dnsResponseBypassPacketInFlow() binding.Flow {
+	dnsPacketInTable := c.pipeline[AntreaPolicyIngressRuleTable]
+	return dnsPacketInTable.BuildFlow(priorityDNSBypass).
+		MatchRegFieldWithValue(CustomReasonField, CustomReasonDNS).
```

Comment:
Could we use an unified reg bit to mark packetout we want it bypass conntrack and policy in the future? I think we don't need the information about the reason the packetout is generated.

Reply:
I think we can. Good point.

Reply:
Added a TODO comment